### ISSUE TITLE
ISSUE-9187: Token authentication not support JSON type auth params as described in doc

### ISSUE THREAD

**Issue by sijie** [streamnative/pulsar#2003]
Original Issue: apache/pulsar#9187 

---

**Describe the bug**
https://pulsar.apache.org/docs/en/security-jwt/#enable-token-authentication-on-brokers
In the official doc, token auth params are configured in JSON type, like:
```properties
brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.AuthenticationToken
brokerClientAuthenticationParameters={"token":"eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0ZXN0LXVzZXIifQ.9OHgE9ZUDeBTZs7nSMEFIuGNEX18FLR3qvy8mqxSxXw"}
```

While in `org.apache.pulsar.client.impl.auth.AuthenticationToken`, it does not support JSON type auth params:
```java
 @Override
    public void configure(String encodedAuthParamString) {
        // Interpret the whole param string as the token. If the string contains the notation `token:xxxxx` then strip
        // the prefix
        if (encodedAuthParamString.startsWith("token:")) {
            this.tokenSupplier = () -> encodedAuthParamString.substring("token:".length());
        } else if (encodedAuthParamString.startsWith("file:")) {
            // Read token from a file
            URI filePath = URI.create(encodedAuthParamString);
            this.tokenSupplier = () -> {
                try {
                    return new String(Files.readAllBytes(Paths.get(filePath)), Charsets.UTF_8).trim();
                } catch (IOException e) {
                    throw new RuntimeException("Failed to read token from file", e);
                }
            };
        } else {
            this.tokenSupplier = () -> encodedAuthParamString;
        }
    }
```
As well as in its test class: `org.apache.pulsar.client.impl.auth.AuthenticationTokenTest`.

**Expected behavior**
The related doc is changed in https://github.com/apache/pulsar/issues/8872 by PR https://github.com/apache/pulsar/pull/8920
I'm not sure it's belong to **doc bug or code bug.**

**Comment by codelipenghui** [comment]
1. double confirm should fix the documentation of the 2.7.0? since the JSON format token does not support in 2.7.0 but the documentation shows the JSON format. @Jennifer88huang 
2. Add support for the JSON format token. @RobertIndie

**Comment by RobertIndie** [comment]
[This PR](https://github.com/apache/pulsar/pull/9313) adds support for the JSON format token.
